test/cli.test.js

```diff
@@ -0,0 +1,70 @@
+var test = require('tape');
+var path = require('path');
+var exec = require('child_process').exec;
+var mvtf = require('@mapbox/mvt-fixtures');
+
+var validate = path.resolve(__dirname, '..', 'bin', 'vtvalidate.js');
+var fixtures = path.resolve(__dirname, '..', 'node_modules', '@mapbox', 'mvt-fixtures');
+
+test('success: valid uncompressed tile', function(t) {
+  var tile = fixtures + '/fixtures/043/tile.mvt';
+  var cmd = [ validate, tile ].join(' ');
+
+  exec(cmd, function(err, stdout, stderr) {
+    t.ifError(err, 'no error');
+    t.equal(stdout, '');
+    t.end();
+  });
+});
+
+test('success: valid gzip compressed tile', function(t) {
+  var tile = fixtures + '/real-world/compressed/14-9384-9577.mvt.gz';
+  var cmd = [ validate, tile ].join(' ');
+
+  exec(cmd, function(err, stdout, stderr) {
+    t.ifError(err, 'no error');
+    t.equal(stdout, '');
+    t.end();
+  });
+});
+
+test('failure: missing tile arg', function(t) {
+  exec(validate, function(err, stdout, stderr) {
+    t.ok(err);
+    t.ok(stdout.indexOf('usage') == 0);
+    t.end();
+  });
+});
+
+test('invalid filetype', function(t) {
+  var tile = fixtures + '/fixtures/043/info.json';
+  var cmd = [ validate, tile ].join(' ');
+
+  exec(cmd, function(err, stdout, stderr) {
+    t.ifError(err, 'no error');
+    t.equal(stdout, 'unknown pbf field type exception\n'); // TODO: figure out where the newline is coming from
+    t.end();
+  });
+});
+
+test('invalid tile arg, not a filepath', function(t) {
+  var tile = 'uhoh';
+  var cmd = [ validate, tile ].join(' ');
+
+  exec(cmd, function(err, stdout, stderr) {
+    t.ok(err);
+    t.ok(stdout.indexOf('uhoh does not exist') == 0);
+    t.end();
+  });
+});
+
+test('invalid tile', function(t) {
+  var tile = fixtures + '/fixtures/003/tile.mvt';
+  var cmd = [ validate, tile ].join(' ');
+
+  exec(cmd, function(err, stdout, stderr) {
+    t.ifError(err, 'no error');
+    t.equal(stdout, 'unknown geometry type\n'); // TODO: figure out where the newline is coming from
```

Comment:
@GretaCB I think `console.log` adds a newline automatically. To avoid this I've used `process.stderr.write` or `process.stdout.write` to avoid the newline. I've not looked but I presume `console.log` uses those under the hood.

Reply:
You could also do `t.equal(stdout.trim(), 'unknown geometry type'); ` to ignore the newline.

Reply:
Indeed: https://stackoverflow.com/questions/34957613/node-js-write-to-stdout-without-a-newline